controller/store/types.go

```diff
@@ -27,23 +27,23 @@ type HAProxySrv struct {
 	Name     string
 	Address  string
 	Modified bool
+	Port     int64
 }
 
 // PortEndpoints describes endpoints of a service port
 type PortEndpoints struct {
 	Port            int64
-	BackendName     string // For runtime operations
 	DynUpdateFailed bool
 	AddrCount       int
 	AddrNew         map[string]struct{}
-	HAProxySrvs     []*HAProxySrv
 }
 
 // Endpoints describes endpoints of a service
 type Endpoints struct {
+	SliceName string
```

Comment:
I think there is no need to have `PortEndpoints` and `Endpoitns` any more.
We used to do that because we wanted to map all addresses to corresponding port but now that we need to work using slices we can just have the following:
```
type Endpoints struct {                                                                               
  SliceName string       
  Namespace string  
  Service        string                                         
  Ports            []int64
  AddrCount   int
  AddrNew     map[string]struct{}
  Status         Status                                        
}
```
And if we want to get the union of all addresses we can use the `getAddressMap` you have just added